lib/sender.ts

```diff
@@ -0,0 +1,323 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as rhea from "rhea";
+import * as log from "./log";
+import { Session } from "./session";
+import { Connection } from "./connection";
+import { SenderEvents } from "rhea";
+import { defaultOperationTimeoutInSeconds } from "./util/constants";
+import { Func } from "./util/utils";
+
+/**
+ * Descibes the options that can be provided while creating an AMQP sender.
+ * @interface SenderOptions
+ */
+export interface SenderOptions extends rhea.SenderOptions {
+  /**
+   * @property {rhea.OnAmqpEvent} [onAccepted] The handler that can be provided for receiving the
+   * "accepted" event after a message is sent from the underlying rhea sender.
+   */
+  onAccepted?: rhea.OnAmqpEvent;
```

Comment:
what's the purpose of having explicit options for all of these event handlers? you already support people registering generic listeners and i would expect most people to be using the return values of the promises for most of these anyway